pkg/cosign/verify.go

```diff
@@ -469,20 +471,32 @@ func bundleHash(bundleBody, signature string) (string, string, error) {
 		return *intotoObj.Content.Hash.Algorithm, *intotoObj.Content.Hash.Value, nil
 	}
 
-	err = json.Unmarshal(bodyDecoded, &rekord)
+	if err := json.Unmarshal(bodyDecoded, &rekord); err == nil {
```

Comment:
did you double check this actually fails? Sometimes unmarshal will succeed even if it's the wrong type. We might need to get a bit more clever here.

Reply:
yep! can confirm that for a hashed rekord signature, it actually travels the second path and doesn't unmarshal to a rekord without an error

Reply:
i just checked how it errors: errors out on "invalid kind value" (rekor checks to make sure that the type we want to unmarshal to is the kind in the JSON)

Reply:
Awesome!